pkg/standalone-utils/contracts/base/VaultPermit.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-vault/contracts/interfaces/IVault.sol";
+import "../interfaces/IPermit.sol";
+import "../interfaces/IPermitDAI.sol";
+
+/**
+ * @notice Functionality to call permit on any EIP2612 token or DAI for interactions with the Vault.
+ * @dev These functions are made payable as they are expected to be delegate called into
+ *      within a multicall which may receive ether.
+ */
+contract VaultPermit {
```

Comment:
I'd consider merging this with `PermitRelayer`: by itself this doesn't add much and I found it harder to understand the whole contract with this added indirection.